timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -259,19 +260,20 @@ def _validate_multi_index_data_frame(cls, data: pd.DataFrame):
 
     @classmethod
     def _construct_pandas_frame_from_iterable_dataset(cls, iterable_dataset: Iterable) -> pd.DataFrame:
-        cls._validate_iterable(iterable_dataset)
-
-        all_ts = []
-        for i, ts in enumerate(iterable_dataset):
+        def load_single_item(item_id: int, ts: dict) -> pd.DataFrame:
             start_timestamp = ts["start"]
             freq = start_timestamp.freq
             if isinstance(start_timestamp, pd.Period):
                 start_timestamp = start_timestamp.to_timestamp(how="S")
             target = ts["target"]
             datetime_index = tuple(pd.date_range(start_timestamp, periods=len(target), freq=freq))
-            idx = pd.MultiIndex.from_product([(i,), datetime_index], names=[ITEMID, TIMESTAMP])
-            ts_df = pd.Series(target, name="target", index=idx).to_frame()
-            all_ts.append(ts_df)
+            idx = pd.MultiIndex.from_product([(item_id,), datetime_index], names=[ITEMID, TIMESTAMP])
+            return pd.Series(target, name="target", index=idx).to_frame()
+
+        cls._validate_iterable(iterable_dataset)
+        all_ts = Parallel(n_jobs=-1)(
```

Comment:
We should allow users to override this from HPs such as `env.num_workers`

Reply:
This code is only executed outside of the `TimeSeriesPredictor` (when user constructs a `TimeSeriesDataframe` with `TimeSeriesDataframe.from_iterable_dataset`), so this method has no access to HPs.

How about we add an optional kwarg `num_cpus: int = -1`  to `TimeSeriesDataframe.from_iterable_dataset` that controls the number of jobs here?

Reply:
yes, adding to `from_iterable_dataset` sounds good to me

Reply:
@tonyhoo Done 👍